File: repose-support/loader-test-filter-one/src/main/scala/org/openrepose/filters/loadertestfilterone/commonfilter/FirstFilter.scala
package org.openrepose.filters.loadertestfilterone.commonfilter

import javax.inject.Named
import javax.servlet.http.HttpServletRequest
import javax.servlet.{FilterChain, FilterConfig, ServletRequest, ServletResponse, Filter}

/**
 * Created by dimi5963 on 11/6/14.
 */
@Named
class FirstFilter extends Filter {
  override def init(p1: FilterConfig): Unit = {
    //Meh?
  }

  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
    val httpRequest: HttpServletRequest = request.asInstanceOf[HttpServletRequest]
    val w: CLServletRequestWrapper = new CLServletRequestWrapper(httpRequest)
    chain.doFilter(w, response)
  }

  override def destroy(): Unit = {

  }
}

File: repose-aggregator/functional-tests/test-bundles/filter-one/src/main/java/org/openrepose/filters/one/FirstFilter/FirstFilter.java
package org.openrepose.filters.one.FirstFilter;

import javax.servlet.*;
import javax.servlet.http.HttpServletRequest;
import java.io.IOException;

/**
 * Created by dimi5963 on 1/5/15.
 */
public class FirstFilter implements Filter {
    @Override
    public void init(FilterConfig filterConfig) throws ServletException {
        //nada...
    }

    @Override
    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
        ServletRequest request = (ServletRequest)new ClassLoaderServletRequestWrapper((HttpServletRequest)servletRequest);
        filterChain.doFilter(request, servletResponse);

    }

    @Override
    public void destroy() {

    }
}
